src/Datadog.Trace/TraceId.cs

```diff
@@ -0,0 +1,212 @@
+            Unsafe.WriteUnaligned(ref lowerBytes[0], randomNumberGenerator.Next());
+
+            var higher = (ulong)BitConverter.ToInt64(higherBytes, startIndex: 0) & 0x7FFFFFFFFFFFFFFF;
+            var lower = (ulong)BitConverter.ToInt64(lowerBytes, startIndex: 0) & 0x7FFFFFFFFFFFFFFF;
+
+            return new TraceId(higher, lower);
+        }
+
+        /// <summary>
+        /// Creates random 64 bit traceId.
+        /// </summary>
+        /// <returns>Instance of randomly generated <see cref="TraceId"/>.</returns>
+        public static TraceId CreateRandom64Bit()
+        {
+            var lowerBytes = new byte[8];
+
+            var randomNumberGenerator = RandomNumberGenerator.Current;
+            Unsafe.WriteUnaligned(ref lowerBytes[0], randomNumberGenerator.Next());
+
+            var lower = (ulong)BitConverter.ToInt64(lowerBytes, startIndex: 0) & 0x7FFFFFFFFFFFFFFF;
+
+            return new TraceId(lower);
+        }
+
+        /// <summary>
+        /// Creates traceId from given 16 or 32 sign string representing traceId in hexadecimal format.
+        /// </summary>
+        /// <param name="id">16 or 32 sign string ID to be parsed.</param>
+        /// <returns>Instance of <see cref="TraceId"/> representing the same traceId as the passed string.</returns>
+        public static TraceId CreateFromString(string id)
+        {
+            try
+            {
+                switch (id.Length)
+                {
+                    case 16:
+                    {
+                        var lower = Convert.ToUInt64(id, fromBase: 16);
+                        return new TraceId(lower);
+                    }
+
+                    case 32:
+                    {
+                        var higherAsString = id.Substring(startIndex: 0, length: 16);
+                        var lowerAsString = id.Substring(startIndex: 16, length: 16);
+
+                        var higher = Convert.ToUInt64(higherAsString, fromBase: 16);
+                        var lower = Convert.ToUInt64(lowerAsString, fromBase: 16);
+
+                        return new TraceId(higher, lower);
+                    }
+
+                    default:
+                    {
+                        return Zero;
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is ArgumentOutOfRangeException || ex is InvalidOperationException || ex is OverflowException || ex is FormatException)
+            {
+                return Zero;
+            }
+        }
+
+        /// <summary>
+        /// Creates traceId from given string representing 64bit traceId in decimal format.
+        /// </summary>
+        /// <param name="id">String ID to be parsed.</param>
+        /// <returns>Instance of 64bit <see cref="TraceId"/> representing the same traceId as the passed string.</returns>
+        public static TraceId CreateFromDecimalString(string id)
```

Comment:
I think it would make sense to name this something like `CreateFromStringDatadogConvention` that way it's clear when it should be used and when it shouldn't be. We can also document in the summary that this 1) Converts to 64-bit and 2) Expects a base 10 number for parsing. What do you think?

Reply:
@zacharycmontoya I went the documentation way. I have changed summaries for the struct itself and its methods, so it is clear how it supports datadog convention. I think this will be easier to change when we want to move on to supporting only 128 bit traceid, as we will just get rid of the methods. Please take a look :)